### PR TITLE
[WIP] GlobalBearTestHelper.py: Add support

### PR DESCRIPTION
**Pull request by sangamcse**
Closes https://github.com/coala/coala/issues/4884

### CODE REVIEW

File: coalib/testing/GlobalBearTestHelper.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        assert isinstance(self, BaseTestHelper)
+        assert isinstance(
+            global_bear, GlobalBear), 'The given bear is not a global bear.'
+        assert isinstance(
+            results, list), 'The given results are not a list.'
+
+        bear_output = get_results(global_bear,
+                                  filename=filename,
+                                  settings=settings)
+        if not check_order:
```
Comment (jayvdb) [inline coalib/testing/GlobalBearTestHelper.py]: not defined

File: coalib/testing/GlobalBearTestHelper.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    """
+
+    def check_results(self,
+                      global_bear,
+                      results,
+                      filename=None,
+                      settings={}):
+
+        assert isinstance(self, BaseTestHelper)
+        assert isinstance(
```
Comment (jayvdb) [inline coalib/testing/GlobalBearTestHelper.py]: use Python 3 annotations on the method arguments, and we have a decorator which will automatically do type checking of them.

**Comment by jayvdb** [comment]
Regarding commit message, please create an issue here describing the enhancement, referring to the cEP, etc.

Do not refer to GSoC in your commit message.  That is irrelevant to the content of the change.

**Comment by sangamcse** [comment]
Still in the process @jayvdb. I will try to cover all the methods and tests and update it asap. Thank you 😄

File: coalib/testing/GlobalBearTestHelper.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+from queue import Queue
+
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.settings.Section import Section
+from coalib.testing.BaseTestHelper import BaseTestHelper
+
+
+def _get_test_path(test_dir,
+                   file):
+    return os.path.join(os.path.dirname(__file__),
```
Comment (sangamcse) [inline coalib/testing/GlobalBearTestHelper.py]: @NiklasMM, I am facing issue with this method. My requirement is the file path of that directory from where this method is called but obviously this written method returns `coalib/testing*` directory.

**Comment by sangamcse** [comment]
No longer working on this.

**Pull request closed without merging**